platform/security/idp/security-idp-client/src/main/java/org/codice/ddf/security/idp/client/AssertionConsumerService.java

```diff
@@ -200,11 +200,16 @@ public Response getSamlResponse(
       @QueryParam(SIG_ALG) String signatureAlgorithm,
       @QueryParam(SIGNATURE) String signature) {
 
-    LOGGER.info("HTTP-Redirect binding should not be used for Single Sign On responses");
+    LOGGER.warn("HTTP-Redirect binding should not be used for Single Sign-On responses");
     if (validateSignature(deflatedSamlResponse, relayState, signatureAlgorithm, signature)) {
       try {
-        return processSamlResponse(
-            RestSecurity.inflateBase64(deflatedSamlResponse), relayState, signature != null);
+        return (StringUtils.isNotBlank(deflatedSamlResponse)
```

Comment:
📘 validation is done here since `validateSignature` can take nulls as inputs and not throw errors

Reply:
✏️ This method is becoming a bit difficult to read. Something like this might be more readable.

```
    LOGGER.warn("HTTP-Redirect binding should not be used for Single Sign-On responses");
    if(StringUtils.isBlank(deflatedSamlResponse)
            || StringUtils.isBlank(relayState)
            || StringUtils.isBlank(signatureAlgorithm))
    {
      return Response.serverError().entity("SAML is in a bad state.").build();
    }

    if(!validateSignature(deflatedSamlResponse, relayState, signatureAlgorithm, signature)) {
      return Response.serverError().entity("Invalid AuthN response signature.").build();
    }
    
    try {
      return processSamlResponse(
              RestSecurity.inflateBase64(deflatedSamlResponse), relayState, signature != null);

    } catch (IOException e) {
      String msg = "Unable to decode and inflate AuthN response.";
      LOGGER.info(msg, e);
      return Response.serverError().entity(msg).build();
    }
```

Reply:
I like this change, thanks for suggesting it!